frontend/src/metabase/visualizations/lib/LineAreaBarPostRender.js

```diff
@@ -252,30 +259,53 @@ function onRenderSetClassName(chart, isStacked) {
   chart.svg().classed("stacked", isStacked);
 }
 
+function getXAxisRotation(chart) {
+  let match = String(chart.settings["graph.x_axis.axis_enabled"] || "").match(
+    /^rotate-(\d+)$/,
+  );
+  if (match) {
+    return -parseInt(match[1], 10);
```

Comment:
just to state the obvious, this is because we want to read eg a 45 degree rotated label from lower left to upper right, rather than having it be read from upper left to lower right?

Reply:
Yeah it's rotating around the right edge of the label so we want to rotate it counter clockwise.

Actually i'm going to change this to return the un-negated rotation and negate it in `onRenderRotateAxis`.

Reply:
I'll add some comments too